web/src/app/administration/workbasket/details/distribution-targets/distribution-targets.component.ts

```diff
@@ -100,19 +107,88 @@ export class DistributionTargetsComponent implements OnChanges, OnDestroy {
 				}
 			});
 
-	}
+    this.orientationSubscription = this.orientationService.getOrientation().subscribe((orientation: Orientation) => {
+      this.calculateNumberItemsList();
+      this.getWorkbaskets();
+    });
+  }
+
+  calculateNumberItemsList() {
+    if (this.panelBody) {
+      const panelBody = this.panelBody.nativeElement.offsetHeight;
+      const cardHeight = 75;
+      const unusedHeight = 200;
+      this.cards = Math.round(((panelBody - unusedHeight) / cardHeight));
+      if (window.innerWidth < 992) { this.cards = Math.floor(this.cards / 2); }
+      this.cards > 0 ? TaskanaQueryParameters.pageSize = this.cards : TaskanaQueryParameters.pageSize = 1;
+    }
+  }
+
+  /**
+   * Get the workbasket
+   * @param side false = left list, true = right list
+   */
+  getWorkbaskets(side?: Side) {
+    if (side === this.side.LEFT) {
+      this.distributionTargetsLeft = [];
+    } else if (side === this.side.RIGHT) {
+      this.distributionTargetsRight = [];
+    } else {
+      this.distributionTargetsLeft = [];
+      this.distributionTargetsRight = [];
+      this.distributionTargetsClone = [];
+    }
+    if (this.distributionTargetsSelected) {
+      TaskanaQueryParameters.pageSize = this.cards + this.distributionTargetsSelected.length;
+    }
+
+    this.workbasketSubscription = this.workbasketService.getWorkBasketsSummary(true,
+      undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined,
+      undefined, undefined, undefined, false).subscribe(
+        (distributionTargetsAvailable: WorkbasketSummaryResource) => {
+          if (side === this.side.LEFT) {
+            this.distributionTargetsLeft = Object.assign([], distributionTargetsAvailable._embedded.workbaskets);
+          } else if (side === this.side.RIGHT) {
+            this.distributionTargetsRight = Object.assign([], distributionTargetsAvailable._embedded.workbaskets);
+          } else {
+            this.distributionTargetsLeft = Object.assign([], distributionTargetsAvailable._embedded.workbaskets);
+            this.distributionTargetsRight = Object.assign([], distributionTargetsAvailable._embedded.workbaskets);
+            this.distributionTargetsClone = Object.assign([], distributionTargetsAvailable._embedded.workbaskets);
+          }
+          this.onRequest(undefined, true);
+          if (!this.pageLeft || !this.pageRight) {
+            this.calculatePages(distributionTargetsAvailable);
+          }
+        });
+  }
+
+  calculatePages(distributionTargetsAvailable) {
+    this.pageLeft = new Page(this.cards,
```

Comment:
We are introducing a lot of complexity into this component, we need to rethink the approach, try to implement infinite scrolling instead of pagination.